packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -647,6 +668,11 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
                     return;
                 }
 
+                // If an encoding is provided, set it on the entry
+                if (options.encoding) {
+                    entry.encoding = options.encoding;
+                }
```

Comment:
Related: please check out my comment above about how setting encoding within the file system could surface some other issues